libs/ml/include/ml/core/node.hpp

```diff
@@ -237,8 +237,10 @@ std::shared_ptr<TensorType> Node<TensorType>::Evaluate(bool is_training)
     {
       throw std::runtime_error("NaN encountered in Node::Evaluate");
     }
-
-    assert(!math::state_overflow<DataType>());
+    if (math::state_overflow<DataType>())
```

Comment:
I'm not convinced that we want to be doing this. What is the justification for switching to runtime overflow checks?

Reply:
My changes were based on  #2237 to avoid a lot of conflicts in ml_model_tests
I'll restore node.hpp

Reply:
this branch was rebased so all unneeded changes are gone